### PR TITLE
feat(angular-v17-primeng): search posts in the blog in a dialog

### LINKED ISSUE #318: Add search functionality with a popup in header angular-v17-PrimeNG
# Angular App version

- [ ] `angular-v17` - has no UI libraries
- [ ] `angular-v17-modMAT` - modules and Angular Material
- [ ] `angular-v17-AnguMAT` - has Angular Material
- [x] `angular-v17-PrimeNG` - has PrimeNG

in Hashnode it's like this 

image

**Describe the solution you'd like**
- a search icon to be added in header
- popup should be adjusted for light and dark theme
- popup dialogs already exist for the follow button in `angular-v17` and `angular-v17-AnguMAT`, use the same folder and implementation but add searching capabilities to search throughout all blog posts

### PR DESCRIPTION
**Pull request by czBalazs98**
# Angular App version

- [ ] `angular-v17` - has no UI libraries
- [ ] `angular-v17-modMAT` - modules and Angular Material
- [ ] `angular-v17-AnguMAT` - has Angular Material
- [x] `angular-v17-PrimeNG` - has PrimeNG

## This PR Closes Issue 
closes #318 

## What type of PR is this? (check all applicable)
- [x] 🅰️ Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Mobile & Desktop Screenshots/Recordings
[Attach screenshots or recordings if applicable]

## Added to documentation?
- [ ] 📜 README.md
- [x] 🙅 no documentation needed

## [Optional] Post-deployment tasks
[Specify any post-deployment tasks that need to be performed]

## [Optional] What gif best describes this PR or how it makes you feel? 
[Embed gif or describe the feeling in plain text]

### CODE REVIEW

**Comment by czBalazs98** [comment]
It's the first version of the implementation of this issue #138. Please check the deployment preview and feel free to give any advice or idea, how do you like this feature this way.

File: angular-v17-PrimeNG/src/app/components/header/header.component.html [hunk 1]
```diff
@@ -4,7 +4,8 @@
       <img src="../../../assets/images/angular-headless-hashnode-logo.jpg" alt="logo" />
       <h1>{{ blogName }}</h1>
     </a>
-    <div class="theme-control">
+    <div class="function-bar">
```
Comment (cleancodecraft) [inline angular-v17-PrimeNG/src/app/components/header/header.component.html:7]: Can we call this class simply `controls` ? 

my suggestion is to try and build upon what is already existing and preserving what we have. All 4 versions of the app have the `theme-control` class surrounding the toggle button. 

Yes, if the search is placed there - it is no longer exactly `heme-control` only, but also - it is not a `function` and it is not a `bar`. 

I suggest `controls` or `actions` class

**Comment by monacodelisa** [comment]
@czBalazs98 I just checked it out and I love how it looks and it works great, great job on tis one 😊👍 This is a very minor detail but I also would like the class to be either `controls` or `actions` - whichever you prefer, just that will make it to maintain uniformity across all app versions, and then I can merge it

**Comment by czBalazs98** [comment]
Thank you for your feedback! I changed the class name to controls :)

**Review by monacodelisa — APPROVED** [review]

**Comment by cleancodecraft** [comment]
@czBalazs98 hey you know what I think - I think this looks better than Hashnode, nicely done 💯

**Review by cleancodecraft — APPROVED** [review]

**Comment by monacodelisa** [comment]
@czBalazs98 merged, thank you :)

**Pull request merged**